spec/models/post_spec.rb

```diff
@@ -0,0 +1,21 @@
+RSpec.describe Post, type: :model do
+    subject { FactoryBot.create(:post) }
+    describe "validation" do
+        it { is_expected.to validate_presence_of(:title) }
+        it { is_expected.to validate_presence_of(:body) }
+
+        it "is invalid with null title" do
+            subject.title = nil
+            expect(subject).to_not be_valid
+        end
+        
+        it "is invalid with null body" do
+            subject.body = nil
+            expect(subject).to_not be_valid
+        end
```

Comment:
It has already been tested above. If you have any custom value that will not be tested by shoulda matchers, then add tests.

Reply:
okay vai, removed and replaced these test cases with empty string test cases